pkg/helpers/helpers.go

```diff
@@ -321,3 +321,69 @@ func ManagedClusterAssetFn(fs embed.FS, managedClusterName string) resourceapply
 		return assets.MustCreateAssetFromTemplate(name, template, config).Data, nil
 	}
 }
+
+func TaintEqual(taint1 clusterv1.Taint, taint2 clusterv1.Taint) bool {
+	// Ignore the comparison of time
+	return taint1.Key == taint2.Key && taint1.Value == taint2.Value && taint1.Effect == taint2.Effect
+}
+
+func IsTaintsEqual(taints1 []clusterv1.Taint, taints2 []clusterv1.Taint) bool {
+	if len(taints1) != len(taints2) {
+		return false
+	}
+	if (taints1 == nil) != (taints2 == nil) {
+		return false
+	}
+	if len(taints2) == 0 && len(taints1) == 0 {
+		return true
+	}
+	for _, v1 := range taints1 {
+		flag := false
+		for _, v2 := range taints2 {
+			if TaintEqual(v1, v2) {
+				flag = true
+				break
+			}
+		}
+		if !flag {
+			return false
+		}
+	}
+
+	return true
+}
+
+func HasTaints(taints []clusterv1.Taint, target clusterv1.Taint) bool {
+	for _, v := range taints {
+		if TaintEqual(v, target) {
+			return true
+		}
+	}
+	return false
+}
+
+func AddTaints(taints *[]clusterv1.Taint, taint clusterv1.Taint) {
```

Comment:
Please return a new array instead of mutating the original one.

Reply:
Also you may need to handle the case in which a taint with same key but different value/effect exists.

Reply:
And you may also return an updated flag to indicate we do the update action